client/v2/autocli/flag/builder.go

```diff
@@ -297,10 +301,18 @@ func (b *Builder) addFlattenFieldBindingToArgs(ctx *context.Context, path string
 
 // addFieldBindingToArgs adds a fieldBinding for a positional argument to the message binder.
 // The fieldBinding is appended to the positional arguments list in the message binder.
-func (b *Builder) addFieldBindingToArgs(ctx *context.Context, messageBinder *MessageBinder, name protoreflect.Name, fields protoreflect.FieldDescriptors) (fieldBinding, error) {
+func (b *Builder) addFieldBindingToArgs(ctx *context.Context, messageBinder *MessageBinder, name protoreflect.Name, fields protoreflect.FieldDescriptors, mustFieldExist bool) (fieldBinding, error) {
 	field := fields.ByName(name)
 	if field == nil {
-		return fieldBinding{}, fmt.Errorf("can't find field %s", name) // TODO: it will improve error if msg.FullName() was included.`
+		if mustFieldExist {
```

Comment:
This should stay as is, to keep the great UX for normal field.
The issue is mainly for inner fields, where we skip if not imported

Reply:
So with this im still getting an issue with zeta, namely:

```sh
panic: can't find field params

goroutine 1 [running]:
main.NewRootCmd()
        /home/julien/downloads/node/cmd/zetacored/root.go:141 +0x70c
main.main()
        /home/julien/downloads/node/cmd/zetacored/main.go:18 +0x20
```

while the field check should still be here: https://github.com/cosmos/cosmos-sdk/pull/24366/files#diff-6bea0895a17d1ff65c8d380fceed137fecb8d4dab095e8cebd9dc8a2c4e9e472L168-L171

Reply:
Turns out the comment was very true, it wasn't due to an inner field as i first thought, but due to a misconfigured autocli config (which should fail as described here). We should definitely improve the error. Trying to do that here as well.

Reply:
Fixed by https://github.com/cosmos/cosmos-sdk/pull/24449/commits/55a10d6e5563c4e152529f687e8d9ed09275ff77. Improved error in https://github.com/cosmos/cosmos-sdk/pull/24449/commits/003cbc980eb4eaa316ab2bb961084d1e6d30705c